pkg/dns/serviceresolver.go

```diff
@@ -128,7 +128,7 @@ func (b *ServiceResolver) Records(dnsName string, exact bool) ([]msg.Service, er
 		}
 
 		// no clusterIP and not headless, no DNS
-		if len(svc.Spec.ClusterIP) == 0 {
+		if len(svc.Spec.ClusterIP) == 0 && svc.Spec.Type != kapi.ServiceTypeExternalName {
```

Comment:
is there any security implication to letting the user control the CNAME that comes back from a DNS query?